ext/http/http_next.rs

```diff
@@ -0,0 +1,801 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+use crate::request_body::HttpRequestBody;
+use crate::request_properties::DefaultHttpRequestProperties;
+use crate::request_properties::HttpConnectionProperties;
+use crate::request_properties::HttpListenProperties;
+use crate::request_properties::HttpPropertyExtractor;
+use crate::response_body::CompletionHandle;
+use crate::response_body::ResponseBytes;
+use crate::response_body::ResponseBytesInner;
+use crate::response_body::V8StreamHttpResponseBody;
+use crate::LocalExecutor;
+use deno_core::error::bad_resource;
+use deno_core::error::AnyError;
+use deno_core::futures::TryFutureExt;
+use deno_core::op;
+use deno_core::serde_v8::Value;
+use deno_core::v8;
+use deno_core::AsyncRefCell;
+use deno_core::BufView;
+use deno_core::ByteString;
+use deno_core::CancelFuture;
+use deno_core::CancelHandle;
+use deno_core::CancelTryFuture;
+use deno_core::OpState;
+use deno_core::RcRef;
+use deno_core::Resource;
+use deno_core::ResourceId;
+use deno_core::ZeroCopyBuf;
+use deno_net::ops_tls::TlsStream;
+use deno_net::raw::put_network_stream_resource;
+use deno_net::raw::take_network_stream_listener_resource;
+use deno_net::raw::take_network_stream_resource;
+use deno_net::raw::NetworkStream;
+use deno_net::raw::NetworkStreamAddress;
+use http::request::Parts;
+use hyper1::body::Incoming;
+use hyper1::header::COOKIE;
+use hyper1::http::HeaderName;
+use hyper1::http::HeaderValue;
+use hyper1::server::conn::http1;
+use hyper1::server::conn::http2;
+use hyper1::service::service_fn;
+use hyper1::upgrade::OnUpgrade;
+use hyper1::StatusCode;
+use pin_project::pin_project;
+use pin_project::pinned_drop;
+use slab::Slab;
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::future::Future;
+use std::io;
+use std::net::Ipv4Addr;
+use std::net::SocketAddr;
+use std::net::SocketAddrV4;
+use std::pin::Pin;
+use std::rc::Rc;
+use tokio::task::spawn_local;
+use tokio::task::JoinHandle;
+
+type Request = hyper1::Request<Incoming>;
+type Response = hyper1::Response<ResponseBytes>;
+
+pub struct HttpSlabRecord {
+  request_info: HttpConnectionProperties,
+  request_parts: Parts,
+  request_body: Option<Incoming>,
+  // The response may get taken before we tear this down
+  response: Option<Response>,
+  body: Option<Rc<HttpRequestBody>>,
+  promise: CompletionHandle,
+  #[cfg(__zombie_http_tracking)]
+  alive: bool,
+}
+
+thread_local! {
+  pub static SLAB: RefCell<Slab<HttpSlabRecord>> = RefCell::new(Slab::with_capacity(1024));
+}
```

Comment:
Likely theoretical, but there is a security concern if users run multiple isolates in the same thread because the slab is shared across isolates.

Reply:
If the user code can access the ops that interact with the slab IDs directly, they have access to much more dangerous ones (most likely code exec). Are ops designed to be secure? If not, I'm pretty sure that an end-user could easily escape the sandbox.

Reply:
The JS functions for ops are exposed to users. The user can call `ops.op_xxxx(...)` with any value they want at any time.

What I meant was that if two isolates are running in the same thread, they share a slab. This can never happen in CLI (I think - @bartlomieju could this happen for tests?)

Reply:
That's correct - it can't happen in the CLI as the tests are run serially and there are no two isolates sharing the same thread. The concern here is actually valid - an embedder could potentially run two isolates on the same thread and hit this problem. This could be solved by storing the slab in the `OpState` but from what I understood it's slower than using a thread local?

Reply:
I think I have `OpState` available everywhere I'm using the slab, but that means chasing a lot of pointer just to get to what we need. This is at least part of why we're so slow serving HTTP. 

Is this just theoretical? Do we actually want to support multiple isolates on one thread, both serving HTTP? That seems like a recipe for extremely poor performance (and we're not testing this anywhere as far as I can see, so other things may just randomly break).

Is there a good reason we don't just say one isolate:one thread and be done with it? It unlocks a lot of performance (and we can do other fun things like putting isolate pointers in thread locals instead of having to pass them around). The OS give us a register dedicated to this magical space and we're missing out on performance if we don't use it.

Reply:
I think, we should bench thread_local vs OpState. `thread_local!` is not zero-cost in Rust and we've seen this before it Tokio too. Ref https://matklad.github.io/2020/10/03/fast-thread-locals-in-rust.html

Reply:
I can follow-up with some more experiments here. If necessary, we could probably drop down one level and use LLVM thread locals rather than Rust ones, or figure out how to store the slab in the isolate itself.